Chase_step1b_AssignOutcome_Algorithm4.sas

```diff
@@ -1,162 +1,205 @@
-
-  *2 - ...and discordant delivery codes;
-   else do;
-     Discordant4=2; Disc_OutPot4="PRD-Dsc"; *has deliv proc but combo of outcomes discordant;
-   end;
-  end;
+	*First, determine if there is a delivery procedure code present;
+  	if put(delivery,$outcdpr.) = 1 then do; *any delivery proc? check for concordant combos;
+
+		*CDL: MODIFIED -- Cleaner than original code but provided the same result among those with discordant4 = 1
+		-- Same modification as Algorithm 3;
+		**step 1b1 part 2;
+	    *1 - ...and concordant delivery codes (redo table 1 check);
+		if lbm NE ('000') /*and udl NE ('000')*/ And SB='000' And LBS='000' and MLS='000' 
+	         then do; discordant4=1; Disc_OutPot4='PRD-LBM';end;
+		else if lbs NE ('000') /*and udl NE ('000')*/ And SB='000' And LBM='000' and MLS='000' 
+	         then do; discordant4=1; Disc_OutPot4='PRD-LBS';end;
+		else if mls NE ('000') /*and udl NE ('000')*/ And SB='000' And LBM='000' and LBS='000'
+	         then do; discordant4=1; Disc_OutPot4='PRD-MLS';end;
+	    else if sb NE ('000') /*and udl NE ('000')*/ And MLS='000' And LBM='000' and LBS='000' 
+	        then do; discordant4=1; Disc_OutPot4='PRD-SB';end;
+		else if UDL NE ('000') and SB In ('000') And MLS='000' And LBM='000' and LBS='000' 
+	        then do; discordant4=1; Disc_OutPot4='PRD-UDL';end;
+
+/*   		*1 - ...concordant delivery procedures (redo table 1 check);*/
+/*    	if lbm Not In ('000') and udl Not In ('000') And SB='000' And LBS='000' and MLS='000' */
+/*         	then do; Discordant4=1; Disc_OutPot4='PRD-LBM';end;*/
+/*    	else if lbs Not In ('000') and udl Not In ('000') And SB='000' And LBM='000' and MLS='000' */
+/*        	then do; Discordant4=1; Disc_OutPot4='PRD-LBS';end;*/
+/*	    else if mls Not In ('000') and udl Not In ('000') And SB='000' And LBM='000' and LBS='000'*/
+/*	         then do; Discordant4=1; Disc_OutPot4='PRD-MLS';end;*/
+/*	    else if sb Not In ('000') and udl Not In ('000') And MLS='000' And LBM='000' and LBS='000' */
+/*	        then do; Discordant4=1; Disc_OutPot4='PRD-SB';end;*/
+/**/
+/*	    *if ignoring non-delivery then single delivery outcome types are concurrent;*/
+/*	    else if lbm Not In ('000') and udl In ('000') And SB='000' And LBS='000' and MLS='000' */
+/*	         then do; Discordant4=1; Disc_OutPot4='PRD-LBM';end;*/
+/*	    else if lbs Not In ('000') and udl In ('000') And SB='000' And LBM='000' and MLS='000' */
+/*	        then do; Discordant4=1; Disc_OutPot4='PRD-LBS';end;*/
+/*	    else if mls Not In ('000') and udl In ('000') And SB='000' And LBM='000' and LBS='000'*/
+/*	         then do; Discordant4=1; Disc_OutPot4='PRD-MLS';end;*/
+/*	    else if sb Not In ('000') and udl In ('000') And MLS='000' And LBM='000' and LBS='000' */
+/*	        then do; Discordant4=1; Disc_OutPot4='PRD-SB';end;*/
+/*	    else if UDL Not In ('000') and SB In ('000') And MLS='000' And LBM='000' and LBS='000' */
+/*	        then do; Discordant4=1; Disc_OutPot4='PRD-UDL';end;*/

```

Comment:
Commented out old code. Cleaner than the original (now commented out) version. Same results - confirmed.